pxtcompiler/emitter/service.ts

```diff
@@ -2146,20 +2148,28 @@ namespace ts.pxtc.service {
         return (n && n.kind === SK.SourceFile) ? null : n;
     }
 
-    function getNameOfWidenedType(t: Type, tc: TypeChecker) {
+    function tsTypeToPxtTypeString(t: Type, tc: TypeChecker) {
         if (t.flags & TypeFlags.NumberLiteral) {
-            return "number";
+            return "Number";
         }
         else if (t.flags & TypeFlags.StringLiteral) {
             return "String";
         }
         else if (t.flags & TypeFlags.BooleanLiteral) {
-            return "boolean";
+            return "Boolean";
         }
 
-        return tc.typeToString(t);
+        const tcString = tc.typeToString(t);
+        const primativeToQname: { [key: string]: string } = {
```

Comment:
not we have a Map type

Reply:
How do we change the policy on this? 
This is standard .ts syntax and I feel it should be prefered to pxt.U.Map. Also, static TS supports { [key: string]: string } syntax but not pxt.U.Map (somewhat obviously)

Not going to hold up this PR on this tho so I'll switch to Map

Reply:
(not/nit auto complete...) Discuss it at the team meeting.